code/modules/power/lighting.dm

```diff
@@ -368,31 +368,35 @@
 			on = FALSE
 	emergency_mode = FALSE
 	if(on)
-		var/BR = brightness
-		var/PO = bulb_power
-		var/CO = bulb_colour
-		if(color)
-			CO = color
-		var/area/A = get_area(src)
-		if (A?.fire)
-			CO = bulb_emergency_colour
-		else if (nightshift_enabled)
-			BR = nightshift_brightness
-			PO = nightshift_light_power
-			if(!color)
-				CO = nightshift_light_color
-		var/matching = light && BR == light.light_range && PO == light.light_power && CO == light.light_color
-		if(!matching)
-			switchcount++
-			if(rigged)
-				if(status == LIGHT_OK && trigger)
-					explode()
-			else if( prob( min(60, (switchcount^2)*0.01) ) )
-				if(trigger)
-					burn_out()
-			else
-				use_power = ACTIVE_POWER_USE
-				set_light(BR, PO, CO)
+		spawn(rand(10,30))	//SKYRAT EDIT ADDITION - AESTHETICS
```

Comment:
Use a timer or sleep() instead here perhaps?

Reply:
This should prevent any edge cases with spamming lights, as they will respect if the switch is OFF or ON, with the check you see, so that lights that are starting up, can still run their code if someone is spamming it. If we used sleep, people could spam the switch and then break lights as the whole code is sleeping. And timers, well, it prevents timer subsystem lag.